src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -270,6 +345,20 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
             return unformattedInterfaceDocument;
         }
 
+        private Document GetUnformattedInterfaceDocument(
+            Solution solution,
+            INamedTypeSymbol extractedInterfaceSymbol,
+            INamedTypeSymbol originalType,
+            DocumentId interfaceDocumentId,
+            CancellationToken cancellationToken)
+        {
+            return CodeGenerator.AddNamedTypeDeclarationAsync(
+                    solution: solution,
+                    destination: originalType.ContainingNamespace,
+                    namedType: extractedInterfaceSymbol,
+                    cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
Why are we doing this?  This is a code-action right, so why can't we be async all the way up?